### PR TITLE
feat: Add support for v2 model_dump and v1 dict to pydantic plugin

### LINKED ISSUE #3147: Enhancement: Support all kwargs for `model_dump` in PydanticPlugin
### Summary

`litestar.contrib.pydantic.PydanticPlugin` has the argument `prefer_alias` which sets the `by_alias` argument of the created Pydantic type encoders. The type encoders are based on `pydantic.BaseModel.model_dump` which also has the arguments `exclude_none` and `exclude_defaults`. This enhancement suggests to add similar arguments to `PydanticPlugin` to configure the type encoders.

### Basic Example

```python
import pydantic
import typing
import litestar
import litestar.contrib.pydantic

class MyModel(pydantic.BaseModel):
    some_number: int = pydantic.Field(alias="someNumber")
    title: str = "a number"
    comment: typing.Optional[str] = None

@litestar.get("/")
def get_stuff() -> MyModel:
    return MyModel(some_number=42)

app = litestar.Litestar(
    route_handlers=[get_stuff],
    plugins=[
        litestar.contrib.pydantic.PydanticPlugin(
            prefer_alias=True,
            # NOTE: The argument names are just provisional
            exclude_none=True,  # new
            exclude_defaults=True,  # new
        )
    ],
)

```
A GET request to `/` would return `{"someNumber":42}` because of the plugin configuration. Without it, it would return `{"some_number":42,"title":"a number","comment":null}`.

### Drawbacks and Impact

You can currently achieve the same result without this feature by adding a custom type encoder for `pydantic.BaseModel`. However, it must be combined with `PydanticPlugin` for the generated schema to use aliases.
```python
import pydantic
import typing
import litestar
import litestar.contrib.pydantic

class MyModel(pydantic.BaseModel):
    some_number: int = pydantic.Field(alias="someNumber")
    title: str = "a number"
    comment: typing.Optional[str] = None

@litestar.get("/")
def get_stuff() -> MyModel:
    return MyModel(some_number=42)


app = litestar.Litestar(
    route_handlers=[get_stuff],
    plugins=[litestar.contrib.pydantic.PydanticPlugin(prefer_alias=True)],
    type_encoders={
        pydantic.BaseModel: lambda x: x.model_dump(mode="json", by_alias=True, exclude_none=True, exclude_default=True),
    },
)

```

Therefore, the only purpose of this enhancement is to make this use case (which I believe to be a common one) simpler to express in code by making the plugin able to do more, reducing the need for type encoders. The community must decide if this is desirable or not.

As long as the default values of the new arguments match the old behavior, this should not cause any issues with backwards compatibility. 

### Unresolved questions

I can see that there may be concerns about having to support a growing list of `pydantic.BaseModel.model_dump` arguments (for example `exlude_unset`). I believe that concern to be valid, but you could always draw a line and say "the plugin supports these extremely common arguments, if you want more, use a custom type encoder". Still, the concern is valid and I believe it should be part of the discussion.

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3147">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3147/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by kedod**
## Description

- Add support for additional V1 model_dump and V2 dict parameters in PydanticPlugin

This is a continuation of https://github.com/litestar-org/litestar/pull/3171
The last unresolved conversation is here https://github.com/litestar-org/litestar/pull/3171#discussion_r1544483848

### CODE REVIEW

**Review by guacs — CHANGES_REQUESTED** [review]

File: litestar/types/serialization.py [hunk 1]
```diff
@@ -57,3 +62,5 @@
 EncodableMsgSpecType: TypeAlias = "Ext | Raw | Struct"
 LitestarEncodableType: TypeAlias = "EncodableBuiltinType | EncodableBuiltinCollectionType | EncodableStdLibType | EncodableStdLibIPType | EncodableMsgSpecType | BaseModel | AttrsInstance"  # pyright: ignore
 DataContainerType: TypeAlias = "Struct | BaseModel | AttrsInstance | TypedDictClass | DataclassProtocol"  # pyright: ignore
+PydanticV2FieldsListType: TypeAlias = "Set[int] | Set[str] | Dict[int, Any] | Dict[str, Any] | None"
```
Comment (Alc-Alc) [inline litestar/types/serialization.py:65]: just for my understanding, is it ok to have ` | None` in the type alias? I would prefer `PydanticV2FieldsListType | None` just to be explicit, but I am not sure if that is the norm 🙂

**Comment by provinzkraut** [comment]
@kedod I started something similar over in #3573, not realising this PR exists :)
Would you mind merging the stuff I did there over here? No need to implement these things twice.

**Comment by Anu-cool-007** [comment]
Hi @provinzkraut, I've created this new PR that combines this one with strict validation: #3608

**Comment by provinzkraut** [comment]
Superseded by #3608

**Pull request closed without merging**